includes/multi-currency/MultiCurrency.php

```diff
@@ -916,6 +916,12 @@ private function get_stored_currency_code() {
 		WC()->initialize_session();
 		$currency_code = WC()->session->get( self::CURRENCY_SESSION_KEY );
 
+		// If the user does not have a session cookie yet and enabled auto-currency selection from geolocation.
+		if ( ! $currency_code && $this->is_using_auto_currency_switching() ) {
+			// Refetch the currency from geolocation.
+			$currency_code = $this->geolocation->get_currency_by_customer_location();
```

Comment:
This code affects similar extensions to the StoreAPI. This means that when this patch is shipped, the price has a different currency when the geolocation is enabled. For this reason, we could break some integrations/workflows that __expect__ to receive the default currency. We should communicate these changes upfront in enough time so that consumers of the data are fully aware of how it may impact them.